staging/src/k8s.io/apiserver/pkg/authentication/config/v1alpha1/types.go

```diff
@@ -0,0 +1,109 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AuthenticationConfiguration struct {
+	metav1.TypeMeta
+
+	// jwt is a list of OIDC providers to authenticate Kubernetes users.
+	// For an incoming token, each JWT authenticator will be attempted in
+	// the order in which it is specified in this list.  Note however that
+	// other authenticators may run before or after the JWT authenticators.
+	// The specific position of JWT authenticators in relation to other
+	// authenticators is neither defined nor stable across releases.  Since
+	// each JWT authenticator must have a unique issuer URL, at most one
+	// JWT authenticator will attempt to cryptographically validate the token.
+	JWT []JWTAuthenticator `json:"jwt"`
+}
+
+type JWTAuthenticator struct {
+	// issuer is a basic OIDC provider connection options.
+	Issuer Issuer `json:"issuer"`
+
+	// claimValidationRules are rules that are applied to validate token claims to authenticate users.
+	// +optional
+	ClaimValidationRules []ClaimValidationRule `json:"claimValidationRules,omitempty"`
+
+	// claimMappings points claims of a token to be treated as user attributes.
+	ClaimMappings ClaimMappings `json:"claimMappings"`
+}
+
+type Issuer struct {
+	// url points to the issuer URL in a format https://url/path.
+	// This must match the "iss" claim in the presented JWT, and the issuer returned from discovery.
+	// Same value as the --oidc-issuer-url flag.
+	// Used to fetch discovery information unless overridden by discoveryURL.
+	// Required to be unique.
+	// Note that egress selection configuration is not used for this network connection.
+	// TODO(aramase): decide if we want to support egress selection configuration and how to do so.
+	URL string `json:"url"`
+
+	// certificateAuthority contains PEM-encoded certificate authority certificates
+	// used to validate the connection when fetching discovery information.
+	// If unset, the system verifier is used.
+	// Same value as the content of the file referenced by the --oidc-ca-file flag.
+	// +optional
+	CertificateAuthority []byte `json:"certificateAuthority,omitempty"`
+
+	// clientIDs is the set of acceptable audiences the JWT must be issued to.
+	// At least one of the entries must match the "aud" claim in presented JWTs.
+	// Same value as the --oidc-client-id flag (though this field supports an array).
+	// Required to be non-empty.
+	ClientIDs []string `json:"clientIDs"`
+}
+
+type ClaimValidationRule struct {
+	// claim is the name of a required claim.
+	// Same as --oidc-required-claim flag.
+	// Only string claims are supported.
+	// +optional
+	Claim string `json:"claim"`
+	// requiredValue is the value of a required claim.
+	// Same as --oidc-required-claim flag.
+	// +optional
+	RequiredValue string `json:"requiredValue"`
```

Comment:
Previously this was a map, now we need to validate that the claim names do not overlap.